bench/mjolnir/buildtiles.cc

```diff
@@ -0,0 +1,55 @@
+#include <benchmark/benchmark.h>
+
+#include "baldr/rapidjson_utils.h"
+#include "filesystem.h"
+#include "mjolnir/util.h"
+
+using namespace valhalla::mjolnir;
+
+namespace {
+
+void BM_SomeFunction(benchmark::State& state, const std::string& pbf_path) {
+  valhalla::midgard::logging::Configure({{"type", ""}});
+
+  std::string const kConfigFilePath = VALHALLA_SOURCE_DIR "bench/mjolnir/config.json";
+  boost::property_tree::ptree config;
+  rapidjson::read_json(kConfigFilePath, config);
+  config.get_child("mjolnir").erase("tile_extract");
+  config.get_child("mjolnir").erase("tile_url");
+
+  BuildStage const start_stage_str = string_to_buildstage("initialize");
+  BuildStage const end_stage_str = string_to_buildstage("parseways");
+
+  for (auto _ : state) {
+    build_tile_set(config, {pbf_path}, start_stage_str, end_stage_str);
+  }
+};
+
+struct FileWithSize {
+  filesystem::path path;
+  uintmax_t size;
+
+  bool operator<(const FileWithSize& file) const {
+    return size < file.size;
+  }
+};
+
+std::vector<FileWithSize> CollectInputProtobufs(const std::string& path) {
+  std::vector<FileWithSize> files;
+  for (filesystem::recursive_directory_iterator i(path), end; i != end; ++i)
+    files.push_back({i->path(), i->file_size()});
+  std::sort(files.begin(), files.end());
```

Comment:
Alternatively, we could use an explicit manifest of the PBF files. There's a similar example here for benchmark fixtures: https://github.com/valhalla/valhalla/blob/master/bench/meili/mapmatch.cc#L97